ci/build_container/build_recipes/zlib.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -e
+
+VERSION=1.2.11
+
+wget -O zlib-$VERSION.tar.gz https://github.com/madler/zlib/archive/v$VERSION.tar.gz
+tar xf zlib-$VERSION.tar.gz
+cd zlib-$VERSION
+./configure
+make V=1 install
+cp libz.a $THIRDPARTY_BUILD/lib
```

Comment:
Hey @mattklein123, I remember you ask me to build an image.  I assume that it's done by running any of the >> ./ci/run_envoy_docker.sh './ci/do_ci.sh ..'
If that's correct, when I added zlib as external_deps, do_ci fails:
```
ERROR: missing input file '@envoy//ci/prebuilt:thirdparty_build/lib/libz.a'.
ERROR: /build/envoy-filter-example/BUILD:10:1: //:envoy: missing input file '@envoy//ci/prebuilt:thirdparty_build/lib/libz.a'.
```